server/lobbyconnection.py

```diff
@@ -830,19 +832,32 @@ async def command_game_join(self, message):
 
         await self.launch_game(game, is_host=False)
 
+    @ice_only
     async def command_game_matchmaking(self, message):
         queue_name = str(
             message.get('queue_name') or message.get('mod', 'ladder1v1')
         )
         state = str(message['state'])
-
-        if self._attempted_connectivity_test:
-            raise ClientError("Cannot host game. Please update your client to the newest version.")
+        party = self.party_service.get_party(self.player)
 
         if state == "stop":
             await self.ladder_service.cancel_search(self.player, queue_name)
             return
 
+        if party is not None:
```

Comment:
Additionally I think we can exit early if the requesting player is not the owner of their party.